src/main/java/fruitshop/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package fruitshop.strategy;
+
+import fruitshop.model.FruitTransaction;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler>
+                                         operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void getOperation(FruitTransaction fruitTransaction) {
+        OperationHandler operationHandler = operationHandlers.get(fruitTransaction.getOperation());
+        operationHandler.handler(fruitTransaction);
```

Comment:
There is a potential for a NullPointerException if the operationHandlers map does not contain a handler for the given fruitTransaction operation. Consider adding a check for null and throw a custom exception with an informative message if the operationHandler is not found.